lib/sla.js

```diff
@@ -102,26 +102,45 @@ function addChannelsToBridge(inbound, dialed, bridge) {
  */
 function originateChannel(client, inbound, bridge) {
   var defer = Q.defer();
-  var dialed = client.Channel();
-  var originate = Q.denodeify(dialed.originate.bind(dialed));
-  originate({endpoint: 'SIP/phone', app: 'sla', appArgs: 'dialed'})
-    .catch(function (err) {
-      err.name = 'originateFailure';
-      defer.reject(err);
-      
+  var dialed = [client.Channel(), client.Channel()];
+  var originate = [];
+  var numToHangup = dialed.length;
+  var station = ['SIP/phone1', 'SIP/phone2'];
+  dialed.forEach(function (channel, index, dialed) {
+    originate.push(Q.denodeify(dialed[index].originate.bind(dialed[index])));
+    originate[index]({endpoint: station[index], app: 'sla', appArgs: 'dialed',
+        timeout: 10})
+      .catch(function (err) {
+        err.name = 'OriginateFailure';
+        defer.reject(err);
+      });
+    dialed[index].once('StasisStart', function(event, line) {
+      var answer = Q.denodeify(line.answer.bind(line));
+      answer();
+      var toKill = dialed.filter(function (unanswered) {
+        return unanswered.id !== line.id;
+      });
+      toKill.forEach(function (unanswered) {
+        var hangup = Q.denodeify(unanswered.hangup.bind(unanswered));
+        hangup();
+      });
+      defer.resolve(line);
+    });
+    inbound.once('ChannelHangupRequest', function(event, line) {
+      defer.resolve('Inbound channel hungup');
+      dialed.forEach(function (unanswered) {
+        var hangup = Q.denodeify(unanswered.hangup.bind(unanswered));
+        hangup();
+      });
+    });
+    dialed[index].once('ChannelDestroyed', function(event) {
+      numToHangup -= 1;
```

Comment:
Why not just remove the channel from the dialed array and check the length instead of having a numToHangup variable to keep track of on top of the array?